PWGCF/TwoParticleCorrelations/Tasks/lambdaR2Correlation.cxx

```diff
@@ -602,24 +598,53 @@ struct lambdaCorrTableProducer {
     if constexpr (reco) {
       auto v0mcpart = v0track.mcParticle();
 
-      histos.fill(HIST("QA_Checks/h1d_tracks_pid_before_mccuts"), v0mcpart.pdgCode(), v0mcpart.pt());
+      histos.fill(HIST("QA_Checks/h2d_tracks_pid_before_mccuts"), v0mcpart.pdgCode(), v0mcpart.pt());
 
       if (cfg_primary_lambda && !v0mcpart.isPhysicalPrimary()) {
         return;
       } else if (cfg_secondary_lambda && v0mcpart.isPhysicalPrimary()) {
         return;
       }
 
-      histos.fill(HIST("QA_Checks/h1d_tracks_pid_phyprim"), v0mcpart.pdgCode(), v0mcpart.pt());
+      histos.fill(HIST("QA_Checks/h2d_tracks_pid_phyprim"), v0mcpart.pdgCode(), v0mcpart.pt());
 
-      if (v0part == kLambda && v0mcpart.pdgCode() != 3122) {
-        return;
-      } else if (v0part == kAntiLambda && v0mcpart.pdgCode() != -3122) {
-        return;
+      // Get Daughters and Mothers
+      bool decay_channel_flag = false;
+      std::vector<int> daughterPDGs{}, daughterIDs{}, motherPDGs{}, motherIDs{};
+      auto mcpart_daughters = v0mcpart.template daughters_as<aod::McParticles>();
+      auto mcpart_mothers = v0mcpart.template mothers_as<aod::McParticles>();
+
+      if (cfg_pid_flag) {
+        if (v0part == kLambda && v0mcpart.pdgCode() != 3122) {
+          return;
+        } else if (v0part == kAntiLambda && v0mcpart.pdgCode() != -3122) {
+          return;
+        }
+
+        getPDGsIDs(mcpart_daughters, daughterPDGs, daughterIDs);
+        getPDGsIDs(mcpart_mothers, motherPDGs, motherIDs);
+
+        // Decay to Proton-Pion
+        if (abs(daughterPDGs[0]) == 2212 && abs(daughterPDGs[1]) == 211) {
+          decay_channel_flag = true;
+        }
+
+        if (!decay_channel_flag) {
+          return;
+        }
+
+        // check the secondary lambdas coming from Sigma, Cascades and Omegas
+        if (abs(motherPDGs[0]) == 3212 || abs(motherPDGs[1]) == 3212) {
+          histos.fill(HIST("QA_Checks/h2d_lambda_from_sigma"), v0mcpart.pdgCode(), v0mcpart.pt());
+        }
+
+        if (abs(motherPDGs[0]) == 3312 || abs(motherPDGs[1]) == 3312) {
+          histos.fill(HIST("QA_Checks/h2d_lambda_from_cascade"), v0mcpart.pdgCode(), v0mcpart.pt());
+        }
```

Comment:
I would suggest start using named constants, as for https://root.cern/doc/master/TPDGCode_8h.html, instead of numbers
Just add
```
#include <TPDGCode.h>
```
for have them available. The code would be more readable and less error prone